contracts/interfaces/IERC7540.sol

```diff
@@ -0,0 +1,102 @@
+        address indexed controller,
+        address indexed owner,
+        uint256 indexed requestId,
+        address sender,
+        uint256 shares
+    );
+
+    event OperatorSet(address indexed controller, address indexed operator, bool approved);
+
+    event DepositProcessed(address indexed controller, uint256 indexed requestId, uint256 amount);
+
+    event RedeemProcessed(address indexed controller, uint256 indexed requestId, uint256 amount);
+
+    // Methods
+
+    /**
+     * @dev Initiates a deposit request.
+     * @param assets The amount of assets to deposit.
+     * @param controller The address of the controller managing the request.
+     * @param owner The owner of the assets.
+     * @return requestId The unique identifier for this deposit request.
+     */
+    function requestDeposit(uint256 assets, address controller, address owner) external returns (uint256 requestId);
+
+    /**
+     * @dev Initiates a redeem request.
+     * @param shares The amount of shares to redeem.
+     * @param controller The address of the controller managing the request.
+     * @param owner The owner of the shares.
+     * @return requestId The unique identifier for this redeem request.
+     */
+    function requestRedeem(uint256 shares, address controller, address owner) external returns (uint256 requestId);
+
+    /**
+     * @dev Gets the pending deposit request amount for a given controller and requestId.
+     * @param requestId The unique identifier for the request.
+     * @param controller The address of the controller.
+     * @return assets The amount of assets in the pending state.
+     */
+    function pendingDepositRequest(uint256 requestId, address controller) external view returns (uint256 assets);
+
+    /**
+     * @dev Gets the claimable deposit request amount for a given controller and requestId.
+     * @param requestId The unique identifier for the request.
+     * @param controller The address of the controller.
+     * @return assets The amount of assets in the claimable state.
+     */
+    function claimableDepositRequest(uint256 requestId, address controller) external view returns (uint256 assets);
+
+    /**
+     * @dev Gets the pending redeem request amount for a given controller and requestId.
+     * @param requestId The unique identifier for the request.
+     * @param controller The address of the controller.
+     * @return shares The amount of shares in the pending state.
+     */
+    function pendingRedeemRequest(uint256 requestId, address controller) external view returns (uint256 shares);
+
+    /**
+     * @dev Gets the claimable redeem request amount for a given controller and requestId.
+     * @param requestId The unique identifier for the request.
+     * @param controller The address of the controller.
+     * @return shares The amount of shares in the claimable state.
+     */
+    function claimableRedeemRequest(uint256 requestId, address controller) external view returns (uint256 shares);
```

Comment:
Ordering could be more consistent. 

Either:
- requests (deposit + redeem)
- pending (deposit + redeem) request
- claimable (deposit + redeem)

Or

- deposit (request + pending request + claimable)
- redeem (request + pending request + claimable)

Reply:
Done @Amxx